src/Blocks/Happypoint.php

```diff
@@ -0,0 +1,137 @@
+                        'type' => 'integer',
+                        'default' => 30,
+                    ],
+                    'iframe_url' => [
+                        'type' => 'string',
+                        'default' => '',
+                    ],
+                    'use_embed_code' => [
+                        'type' => 'boolean',
+                    ],
+                    'embed_code' => [
+                        'type' => 'string',
+                        'default' => '',
+                    ],
+                    'override_default_content' => [
+                        'type' => 'boolean',
+                        'default' => 'false',
+                    ],
+                    'local_content_provider' => [
+                        'type' => 'string',
+                        'default' => 'none',
+                    ],
+                ],
+            ]
+        );
+
+        add_action('rest_api_init', [ self::class, 'register_endpoints' ]);
+    }
+
+    /**
+     * Required by the `BaseBlock` class.
+     *
+     * @param array $fields Unused, required by the abstract function.
+     * @phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter.UnusedParameter
+     */
+    public function prepare_data(array $fields): array
+    {
+        return [];
+    }
+    // @phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter.UnusedParameter
+
+    /**
+     * Get the required data for the frontend.
+     *
+     * @param object $fields This object contains image search params, such as `id`.
+     *
+     * @return array Image data.
+     */
+    public static function get_data(object $fields): array
+    {
+        $options = get_option('planet4_options');
+        $image_id = $fields['id'] ? $fields['id'] : $options['happy_point_bg_image_id'];
+        $img_meta = wp_get_attachment_metadata($image_id);
+        $image_alt = get_post_meta($image_id, '_wp_attachment_image_alt', true);
+
+        $data = [];
+        $data['background_src'] = wp_get_attachment_image_src($image_id, 'retina-large')[0] ?? false;
+        $data['background_srcset'] = wp_get_attachment_image_srcset($image_id, 'retina-large', $img_meta);
+        $data['background_sizes'] = wp_calculate_image_sizes('retina-large', null, null, $image_id);
+        $data['default_content_provider'] = $options['happy_point_content_provider'] ?? 'iframe_url';
+        $data['engaging_network_id'] = $options['engaging_network_form_id'] ?? '';
+        $data['default_image'] = get_bloginfo('template_directory') . '/images/happy-point-block-bg.jpg';
+        $data['background_alt'] = empty($image_alt) ? __('Background image', 'planet4-blocks') : $image_alt;
+        $data['default_embed_code'] = $options['happy_point_embed_code'] ?? '';
+
+        return $data;
+    }
+
+    /**
+     * Endpoint to retrieve the data for the Happypoint block
+     */
+    public static function register_endpoints(): void
+    {
+        register_rest_route(
```

Comment:
Comparing with [this code from the plugin](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/rest/class-rest-api.php#L217-L235), it's better if we pass all the params instead (including the `$id`), since when the `$id` is undefined, then the query is not correctly formatted and will try to get not included `params['id']`.